lib/mixin.js

```diff
@@ -0,0 +1,33 @@
+'use strict';
+
+function arrayUnion(a, b) {
+  let map = {};
+  let i = 0;
+  for (; i < a.length; i++) {
+    map[a[i]] = true;
+  }
+  for (i = 0; i < b.length; i++) {
+    map[b[i]] = true;
+  }
+  return Object.keys(map);
+}
+
+module.exports = function mixin(target, parent) {
+  arrayUnion(Object.keys(target), Object.keys(parent)).forEach(function(key) {
+    const targetValue = target[key];
+    const parentValue = parent[key];
+    
+    // If both the object and the parent define the same key as a function
+    if (typeof targetValue === 'function' && typeof parentValue === 'function') {
+      if (targetValue.toString().indexOf('_super') > -1) {
+        target[key] = function mixinSuperWrapper() {
+          this._super = parent;
+          return targetValue.apply(this, arguments);
+        };
+      }
+    } else if (typeof parentValue !== 'undefined' && !(key in target)) {
```

Comment:
Sorry for being nit-picky, this seems to be getting very close though, so thanks for iterating with me.

I believe the only part of this conditional we should have is`!(key in target)` which suggests the key is only in the parent, and regardless of what its value is even `undefined` we should set it on target.

Example:

```js
let bar = {};
let foo = { member: undefined };
mixin(bar, foo);

'member' in bar // => false (as implemented today)
'member' in bar // => true (as I believe inheritance suggests)
```

Thoughts?